flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/wmassigners/WatermarkAssignerOperator.java

```diff
@@ -190,4 +217,35 @@ public void close() throws Exception {
         FunctionUtils.closeFunction(watermarkGenerator);
         super.close();
     }
+
+    private boolean isIdlenessEnabled() {
+        return idleTimeout > 0;
+    }
+
+    @VisibleForTesting
+    static long calculateProcessingTimeTimerInterval(long watermarkInterval, long idleTimeout) {
+        if (watermarkInterval <= 0) {
+            return idleTimeout;
+        }
+        if (idleTimeout <= 0) {
+            return watermarkInterval;
+        }
```

Comment:
Could we add a `checkArgument(watermarkInterval > 0 || idleTimeout > 0)` in the beginning of this method?

IIUC, we expect the `calculateProcessingTimeTimerInterval` method is only called when `watermarkInterval > 0 || idleTimeout > 0`, and the timerInterval should be > 0. If both of them are 0, the timerInterval will be 0 (It's unexpected).

Currently, `calculateProcessingTimeTimerInterval` is called by only one caller in this PR. And caller has already checked `watermarkInterval > 0 || idleTimeout > 0`, but I'm afraid other callers call `calculateProcessingTimeTimerInterval` in the future, and they doesn't check `watermarkInterval > 0 || idleTimeout > 0`.

It can prevent unexpected call in the future if we add a `checkArgument(watermarkInterval > 0 || idleTimeout > 0)` in the beginning of this method. 

WDYT?